tests/test_jointcalRunner.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of jointcal.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import inspect
+import os.path
+import unittest
+import unittest.mock
+
+import lsst.log
+import lsst.utils
+
+import lsst.daf.persistence
+import lsst.jointcal
+from lsst.meas.algorithms import DatasetConfig
+import lsst.pipe.base
+
+
+# for MemoryTestCase
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class TestJointcalRunner(lsst.utils.tests.TestCase):
+    """Test that JointcalRunner calls JointcalTask with appropriate arguments."""
+    @classmethod
+    def setUpClass(cls):
+        cls.data_dir = os.path.join(lsst.utils.getPackageDir('jointcal'), 'tests/data')
```

Comment:
Why do you prefer to put it at the top of the file instead of in `setUpClass`? `setUpClass` is where "do this once" things go.

Reply:
Both places work, but I tend to prefer putting calculations at file scope that are fixed for the file and I would use `setUpClass` or `setUp` to calculate values derived from that fixed global.

Reply:
Why do you prefer using `__file__` to `getPackageDir`? The former requires three nested `os.path.blah` things.

Reply:
Because I don't believe that a third party package should be used to find some test data that is in the directory below the test script itself. I also think that if I run this test script in isolation then it should find the data related to this version of the test script and not the data relative to some EUPS package setup that might be not what I intend.